pkg/tstune/backup_test.go

```diff
@@ -0,0 +1,182 @@
+	correctFile1 := path.Join(os.TempDir(), "timescaledb_tune.backup201901181122")
+	correctFile2 := path.Join(os.TempDir(), "timescaledb_tune.backup201901191200")
+	cases := []struct {
+		desc        string
+		onDiskFiles []string
+		globErr     bool
+		want        []string
+		errMsg      string
+	}{
+		{
+			desc:        "error on glob",
+			onDiskFiles: []string{"foo"},
+			globErr:     true,
+			errMsg:      errGlob,
+		},
+		{
+			desc:        "no matching files",
+			onDiskFiles: []string{},
+			want:        []string{},
+		},
+		{
+			desc:        "invalid file",
+			onDiskFiles: []string{"foo"},
+			want:        []string{},
+		},
+		{
+			desc:        "one correct file",
+			onDiskFiles: []string{correctFile1},
+			want:        []string{correctFile1},
+		},
+		{
+			desc:        "two correct files",
+			onDiskFiles: []string{correctFile1, correctFile2},
+			want:        []string{correctFile1, correctFile2},
+		},
+		{
+			desc:        "two correct files with wrong files",
+			onDiskFiles: []string{"foo", correctFile1, "bar", correctFile2, "baz"},
+			want:        []string{correctFile1, correctFile2},
+		},
+	}
+	oldFilepathGlobFn := filepathGlobFn
+	for _, c := range cases {
+		filepathGlobFn = func(_ string) ([]string, error) {
+			if c.globErr {
+				return nil, fmt.Errorf(errGlob)
+			}
+			return c.onDiskFiles, nil
+		}
+
+		files, err := getBackups()
+		if c.errMsg == "" && err != nil {
+			t.Errorf("%s: unexpected error: got %v", c.desc, err)
+		} else if c.errMsg != "" {
+			if err == nil {
+				t.Errorf("%s: unexpected lack of error", c.desc)
+			}
+			if got := err.Error(); got != c.errMsg {
+				t.Errorf("%s: unexpected error msg: got\n%s\nwant\n%s", c.desc, got, c.errMsg)
+			}
+		}
+		if got := len(files); got != len(c.want) {
+			t.Errorf("%s: incorrect size of files: got %d want %d", c.desc, got, len(c.want))
+		} else {
+			for i, wantFile := range c.want {
+				if got := files[i]; got != wantFile {
+					t.Errorf("%s: incorrect file at index %d: got %s want %s", c.desc, i, got, wantFile)
+				}
+			}
+		}
+	}
+	filepathGlobFn = oldFilepathGlobFn
+}
+
+func TestFSRestorer(t *testing.T) {
+	fileContents := []byte("oneline\ntwoline\nthreeline\n")
+	tmpfile, err := ioutil.TempFile("", "timescaledb-tune-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(tmpfile.Name()) // clean up
+
+	if _, err = tmpfile.Write(fileContents); err != nil {
+		t.Fatal(err)
+	}
+	if err = tmpfile.Close(); err != nil {
+		t.Fatal(err)
+	}
+	backupPath := tmpfile.Name()
+	tmpfile2, err := ioutil.TempFile("", "timescaledb-tune-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(tmpfile2.Name()) // clean up
+
+	confPath := tmpfile2.Name()
+
+	r := &fsRestorer{}
+	err = r.Restore("", "")
+	if err == nil {
+		t.Fatalf("expected an error 1")
+	}
+	err = r.Restore(backupPath, "")
+	if err == nil {
+		t.Fatalf("expected an error 2")
+	}
+	err = r.Restore(backupPath, confPath)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
```

Comment:
This ensures that no errors occur but doesn't necessarily test that the file is properly restored. Could we potentially write to distinct values to each temp file and read the final conf file to ensure that the code is indeed restoring to the correct file?

Reply:
Ooops, yeah I forgot to do that. Added.